src/components/App/App.jsx

```diff
@@ -28,6 +34,64 @@ import Api from '../Api';
 import Footer from '../Footer';
 import constants from '../constants';
 import FourOhFour from '../../components/FourOhFour';
+import { heroSelector } from '../../reducers/selectors';
+
+const path = '/notifications';
+
+firebase.initializeApp({
+  messagingSenderId: '94888484309',
+});
+
+const messaging = firebase.messaging();
+
+function getMessagingToken() {
+  messaging.getToken()
+    .then((token) => {
+      if (token) {
+        fetch(`${process.env.REACT_APP_API_HOST}${path}`, {
+          credentials: 'include',
+          method: 'POST',
+          headers: {
+            Accept: 'application/json',
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({
+            token,
+          }),
+        })
+          .then((res) => {
+            if (!res.ok) {
+              throw Error();
+            }
+          })
+          .catch(err => console.log(err));
+      }
+    }).catch((err) => {
+      console.log('An error occurred while retrieving token. ', err);
+    });
+}
+
+messaging.onTokenRefresh(getMessagingToken);
+
+messaging.onMessage((payload) => {
+  console.log('Message received. ', payload);
+
+  if (payload.data.type === 'MATCH') {
+    const n = new Notification(
+      'Parsed $match'.replace('$match', payload.data.match_id),
+      {
+        body: 'Check out your performance.',
```

Comment:
localize?

Reply:
I don't know how. I was going to ask you. Strings are only within App, and this is an event listener.

Reply:
maybe you could mount this event listener within componentDidMount in App?